lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -1344,11 +1346,14 @@ sub create_asset {
               && die Mojo::Exception->new("Checksum mismatch expected $sum got: $real_sum ( weak check on last bytes )")
               unless $sum eq $real_sum;
 
+            # finalize upload
             $temp_final_file->move_to($final_file);
-
             chmod 0644, $final_file;
-
             $temp_chunk_folder->remove_tree;
+
+            # adapt job settings to match the adjusted file name
+            $self->settings->search({value => $original_file_name})->update({value => $adjusted_file_name})
+              if defined $adjusted_file_name;
```

Comment:
Is it the right job setting? Shouldn't this be about child jobs settings?
Also, what about restarting parent job? Then, the asset name will be already adjusted so, this won't update it. Maybe it should look for an asset matching wither original name or `[0-9*]-ORIGINAL_NAME`?

Reply:
> Is it the right job setting? Shouldn't this be about child jobs settings?

Right. Considering that I suppose I should scrap this approach again.

Reply:
How hard would be get parent job id into settings? There are already some that are generated, like NAME. Having parent in settings would allow checking both original and renamed assets.